File: admin/src/main/java/com/imstargg/admin/domain/Brawler.java
package com.imstargg.admin.domain;

import com.imstargg.storage.db.core.MessageCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.BrawlStarsImageCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.BrawlerCollectionEntity;
import jakarta.annotation.Nullable;

import java.util.List;

public record Brawler(
        BrawlerCollectionEntity entity,
        List<MessageCollectionEntity> names,
        @Nullable BrawlStarsImageCollectionEntity image,
        List<Gadget> gadgets,
        List<StarPower> starPowers,
        List<Gear> gears
) {
}

File: admin/imstargg-admin-frontend/src/model/Brawler.ts
import {BrawlerEntity} from "@/model/entity/BrawlerEntity";
import {MessageEntity} from "@/model/entity/MessageEntity";
import {BrawlStarsImageEntity} from "@/model/entity/BrawlStarsImageEntity";
import {Gadget} from "@/model/Gadget";
import {StarPower} from "@/model/StarPower";
import {Gear} from "@/model/Gear";

export interface Brawler {
  entity: BrawlerEntity;
  names: MessageEntity[];
  image: BrawlStarsImageEntity | null;
  gadgets: Gadget[];
  starPowers: StarPower[];
  gears: Gear[];
}